TheAlienThatEatsTheCarrot/PhysicsEngine/PhysicsBody.swift

```diff
@@ -93,3 +93,27 @@ extension PhysicsBody: Hashable {
         hasher.combine(ObjectIdentifier(self))
     }
 }
+
+extension PhysicsBody {
+    func isCollidingWith(_ other: PhysicsBody, on direction: Direction) -> Bool {
+        let collisionPoints = self.collider.checkCollision(with: other.collider)
+        if !collisionPoints.hasCollision {
+            return false
+        }
+        let collisionNormal = self.position - other.position
```

Comment:
<img width="577" alt="image" src="https://github.com/cs3217-2324/group-project-the-alien-that-eats-the-carrot/assets/111076731/a7a73540-20ca-4e10-9437-47d4cea2ed11">
can we use this collisionPoints.normal instead?